### PR TITLE
Add CDI Devices to Device Plugin API

### PR DESCRIPTION
**Pull request by elezar**
This change expands the documentation for the Device Plugin to include
the addition of support for CDI Device Names as a feature-gated feature
in the 1.28 release as part of KEP-4009.

* KEP tracking issue: https://github.com/kubernetes/enhancements/issues/4009
* Implementation PR: https://github.com/kubernetes/kubernetes/pull/118254

### CODE REVIEW

**Review by Vikash-8090-Yadav — APPROVED** [review]
LGTM :+1:

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 1]
```diff
@@ -147,6 +147,22 @@ The general workflow of a device plugin includes the following steps:
...
+   * device nodes
+   * environment variables
+   * mounts
+   * fully-qualified CDI device names
+
+   {{< note >}}
+   The processing of the fully-qualified CDI device names by the Device Manager requires
+   the `DevicePluginCDIDevices` feature gate to be enabled. This was added as an alpha feature in
+   v1.28.
+   {{< note >}}
```
Comment (mengjiao-liu) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:164]:
```suggestion
   {{< /note >}}
```

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 2]
```diff
@@ -147,6 +147,22 @@ The general workflow of a device plugin includes the following steps:
    runtime configurations for accessing the allocated devices. The kubelet passes this information
    to the container runtime.
 
+   An `AllocateResponse` contains zero or more `ContainerAllocateResponse` objects. In these, the
+   device plugin defines modifications that must be made to a container's definition to provide
+   access to the device. These modifications include:
+
+   * annotations
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:154]: Are these [Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) or something else?

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 3]
```diff
@@ -147,6 +147,22 @@ The general workflow of a device plugin includes the following steps:
...
+   * device nodes
+   * environment variables
+   * mounts
+   * fully-qualified CDI device names
+
+   {{< note >}}
+   The processing of the fully-qualified CDI device names by the Device Manager requires
+   the `DevicePluginCDIDevices` feature gate to be enabled. This was added as an alpha feature in
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:162]: We'd prefer something like
```diff
-   the `DevicePluginCDIDevices` feature gate to be enabled. This was added as an alpha feature in
+   that the `DevicePluginCDIDevices` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+  is enabled for the kubelet and the kube-apiserver. This was added as an alpha feature in Kubernetes
```

**Review by sftim — COMMENTED** [review]
Thanks for the PR!

Fixups welcome (see inline feedback)

LGTM from https://github.com/kubernetes/website/pull/42001#pullrequestreview-1531753844
and for docs:
/lgtm

---

/approve

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md (hunk 2 above)
Comment (elezar) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:154]: Yes, I think this link is valid. I've updated the link in #42600

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md (hunk 3 above)
Comment (elezar) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:162]: Updated in #42600

**Pull request merged**